include/core/shape.h

```diff
@@ -603,31 +621,72 @@ class DeathStar : public Shape<dim>
 };
 
 /**
- * @class Composite Shapes are currently used only to output the signed distance
- * of particles in the GLS Sharp Navier Stokes solver. The class was however
- * designed so that specific composite shapes could be defined through the
- * parameter file, although this functionality has not been implemented yet.
+ * @class This class was designed so that specific composite shapes could be
+ * defined through the parameter file. Boolean operations such as union,
+ * difference, and intersection are allowed.
  * @tparam dim Dimension of the shape
  */
 template <int dim>
 class CompositeShape : public Shape<dim>
 {
 public:
+  enum class BooleanOperation : int
+  {
+    Union,
+    Difference,
+    Intersection,
+  };
+
   /**
    * @brief Constructs an assembly of shapes into a composite shape
    * @param components The shapes from which this composite sphere will be composed
    */
-  CompositeShape<dim>(std::vector<std::shared_ptr<Shape<dim>>> components)
-    : Shape<dim>(0.,
-                 components[0]->get_position(),
-                 components[0]->get_orientation())
+  CompositeShape<dim>(
+    std::map<unsigned int, std::shared_ptr<Shape<dim>>> components,
```

Comment:
I don,t think component is a good name here because we use components for vectors components.
How about constituent? item?  part?

Reply:
I chose component because of the relationship between component&composite, but I agree it could be confusing. How about component_shapes? What you suggested could also work